File: ngraph/frontend/tensorflow/src/exceptions.cpp
// Copyright (C) 2021 Intel Corporation
// SPDX-License-Identifier: Apache-2.0
//

#include "exceptions.hpp"
#include "node_context.hpp"

namespace ov {
namespace frontend {
namespace tf {
std::string OpValidationFailureTF::get_error_msg_prefix_tf(const tf::NodeContext& node) {
    std::stringstream ss;
    ss << "While validating node '" << node.get_op_type() << '\'';
    return ss.str();
}
}  // namespace tf
}  // namespace frontend
}  // namespace ov

File: src/bindings/python/src/compatibility/ngraph/exceptions.py
# Copyright (C) 2018-2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0

"""ngraph exceptions hierarchy. All exceptions are descendants of NgraphError."""


class NgraphError(Exception):
    """Base class for Ngraph exceptions."""


class UserInputError(NgraphError):
    """User provided unexpected input."""


class NgraphTypeError(NgraphError, TypeError):
    """Type mismatch error."""
